### PR TITLE
Setup tests for release-1.16

### PR DESCRIPTION
**Pull request by wallrj**
As described in the section "Proceed to the post-release "testing and release" steps:" in https://cert-manager.io/docs/contributing/release-process/#minor-releases

Also a couple of drive by fixes.

### CODE REVIEW

File: config/prowgen/prowspecs/specs.go [hunk 1]
```diff
@@ -108,12 +108,32 @@ var knownBranches map[string]BranchSpec = map[string]BranchSpec{
 
 		primaryKubernetesVersion: "1.30",
 
-		// TODO: test k8s 1.31 here when possible; requires support in the release-1.15 branch on cert-manager
 		otherKubernetesVersions: []string{"1.25", "1.26", "1.27", "1.28", "1.29", "1.31"},
```
Comment (wallrj) [inline config/prowgen/prowspecs/specs.go:111]: Comment was out of date. 
1.31 was already added in https://github.com/cert-manager/testing/pull/1060
 * https://github.com/cert-manager/testing/pull/1060

File: config/config.yaml [hunk 2]
```diff
@@ -99,13 +99,20 @@ branch-protection:
...
+                - pull-cert-manager-release-1.16-e2e-v1-31-upgrade
             master:
               required_status_checks:
                 contexts:
                 - pull-cert-manager-master-make-verify
                 - pull-cert-manager-master-make-test
-                - pull-cert-manager-master-e2e-v1-30
-                - pull-cert-manager-master-e2e-v1-30-upgrade
+                - pull-cert-manager-master-e2e-v1-31
+                - pull-cert-manager-master-e2e-v1-31-upgrade
```
[the remark this answers is not in the document]
Comment (SgtCoDFish) [reply config/config.yaml:115]: That's my bad, sorry! Should've updated that in a previous PR!

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

This looks like what I'd have expected. Thanks 🚀

**Comment by wallrj** [comment]
Here's an example of it working:
 * https://github.com/cert-manager/cert-manager/pull/7309#issuecomment-2376903760

**Pull request merged**